offer-manager-domain/src/main/groovy/life/qbic/business/offers/content/CreateOfferContent.groovy

```diff
@@ -0,0 +1,209 @@
+
+        OfferContent offerContentFinal = offerContent.build()
+
+        output.createdOfferContent(offerContentFinal)
+    }
+
+    /**
+     * Helper method that calculates the NET price for a list of product items
+     * @param offerItems The product item list for which the NET is calculated
+     *  @return The net value for the given list of items
+     */
+    private static double calculateNetSum(List<OfferItem> offerItems) {
+        double netSum = 0
+        offerItems.each {
+            netSum += it.quantity * it.unitPrice
+        }
+        return netSum
+    }
+
+    /**
+     * Calculates the overhead sum of all product items
+     * @param offerItems Items for which the overheads are calculated
+     * @return The overhead price for all items
+     */
+    private double calculateOverheadSum(List<OfferItem> offerItems) {
+        double overheadSum = 0.0
+        offerItems.each {
+            overheadSum += it.quantity * it.unitPrice * overheadRatio
+        }
+        return overheadSum
+    }
+
+
+    private OfferItem createProductItemToOfferItem(ProductItem productItem){
+        Product product = productItem.product
+        double unitPrice = (affiliationCategory == AffiliationCategory.INTERNAL) ? product.internalUnitPrice : product.externalUnitPrice
+        double totalCost = unitPrice * productItem.quantity
+
+        OfferItem offerItem = new OfferItem.Builder(productItem.quantity, product.description, product.productName, unitPrice, productItem.quantityDiscount,
+                product.serviceProvider.name(), product.unit.name(),totalCost).build()
+
+        return offerItem
+    }
+
+
+    /**
+     * Adds the product items to the respective product group list
+     * @param offerItems List of productItems contained in offer
+     */
+    private void groupProductItems(List<ProductItem> offerItems) {
+
+        dataGenerationItems = []
+        dataAnalysisItems = []
+        dataManagementItems = []
+
+        // Sort ProductItems into "DataGeneration", "Data Analysis" and "Project & Data Management"
+        offerItems.each {
+            if (it.product.class in productGroupClasses[ProductGroup.DATA_GENERATION]) {
+                dataGenerationItems.add(it)
+            }
+            if (it.product.class in productGroupClasses[ProductGroup.DATA_ANALYSIS]) {
+                dataAnalysisItems.add(it)
+            }
+            if (it.product.class in productGroupClasses[ProductGroup.PROJECT_AND_DATA_MANAGEMENT]) {
+                dataManagementItems.add(it)
+            }
+        }
+    }
+
+    private static HashMap<ProductGroup, List> setProductGroupMapping() {
+        Map<ProductGroup, List> map = [:]
+        map[ProductGroup.DATA_GENERATION] = [Sequencing, ProteomicAnalysis, MetabolomicAnalysis]
+        map[ProductGroup.DATA_ANALYSIS] = [PrimaryAnalysis, SecondaryAnalysis]
+        map[ProductGroup.PROJECT_AND_DATA_MANAGEMENT] = [ProjectManagement, DataStorage]
+
+        return map
+    }
```

Comment:
Here I would set up three lists as static properties instead.
```suggestion
    private static List<Class> DATA_GENERATION_CLASSES = [Sequencing, ProteomicAnalysis, MetabolomicAnalysis]
    private static List<Class> DATA_ANALYSIS_CLASSES = [PrimaryAnalysis, SecondaryAnalysis]
    private static List<Class> PROJECT_AND_DATA_MANAGEMENT_CLASSES = [ProjectManagement, DataStorage]
```

Reply:
+1

Added it by hand to the top of the class